### ISSUE TITLE
Following restart node continues to log "Commitlog shutdown complete" messages

### ISSUE THREAD

**Issue by KnifeyMoloko** [scylladb/scylladb#11508]
## Installation details

Kernel Version: 5.15.0-1019-aws
Scylla version (or git commit hash): `2022.2.0~rc1-20220902.a9bc6d191031` with build-id `074a0cb9e6a5ab36ba5e7f81385e68079ab6eeda`
Cluster size: 5 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- longevity-mv-si-4d-2022-2-db-node-371f29c2-7 (18.202.36.168 | 10.4.10.185) (shards: 14)
- longevity-mv-si-4d-2022-2-db-node-371f29c2-6 (34.243.200.187 | 10.4.8.244) (shards: 14)
- longevity-mv-si-4d-2022-2-db-node-371f29c2-5 (54.216.151.114 | 10.4.9.31) (shards: 14)
- longevity-mv-si-4d-2022-2-db-node-371f29c2-4 (34.249.121.120 | 10.4.9.199) (shards: 14)
- longevity-mv-si-4d-2022-2-db-node-371f29c2-3 (54.217.149.108 | 10.4.11.117) (shards: 14)
- longevity-mv-si-4d-2022-2-db-node-371f29c2-2 (63.34.22.20 | 10.4.11.206) (shards: 14)
- longevity-mv-si-4d-2022-2-db-node-371f29c2-1 (3.249.105.79 | 10.4.8.237) (shards: 14)

OS / Image: `ami-03d21402486bbce67` (aws: eu-west-1)

Test: `longevity-mv-si-4days-test`
Test id: `371f29c2-5d87-4a5a-a487-a974ebf7edd1`
Test name: `enterprise-2022.2/longevity/longevity-mv-si-4days-test`
Test config file(s):

- [longevity-mv-si-4days.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/4d64ffd58cef80f21ea3c4c3e97040bf4cb1ebe4/test-cases/longevity/longevity-mv-si-4days.yaml)

## Issue description

**>>>>>>>**
During `RollingConfigChangeInternodeCompression` we change the internode compression settings (duh!) and restart the nodes in the cluster in a rolling fashion. 

During the shutdown process we get the expected log line informing about the commitlog shutdown:

**node 1 db log:**
```
Sep 07 10:57:44 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[7791]:  [shard  3] commitlog - Commitlog shutdown complete
Sep 07 10:57:44 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[7791]:  [shard  5] commitlog - Commitlog shutdown complete
Sep 07 10:57:44 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[7791]:  [shard 11] commitlog - Commitlog shutdown complete
```

However this message gets logged after the restart every few seconds:

**node 1 db log**:
```
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  5] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  2] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard 11] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard 11] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  4] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  7] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard 11] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard 13] commitlog - Commitlog shutdown complete
Sep 07 10:59:57 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard 13] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  8] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard 12] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  8] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  8] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard 12] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  3] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  2] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  4] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  2] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  1] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  1] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  9] commitlog - Commitlog shutdown complete
Sep 07 11:00:07 longevity-mv-si-4d-2022-2-db-node-371f29c2-1 scylla[11779]:  [shard  9] commitlog - Commitlog shutdown complete
```

This continues on all the db nodes, through 7+ hours of the test, taking up ~85% of the db nodes log output:
```
╰─○ cat system.log | wc -l && cat system.log | grep "Commitlog shutdown complete" | wc -l
338698
288716

```

**<<<<<<<**

- Restore Monitor Stack command: `$ hydra investigate show-monitor 371f29c2-5d87-4a5a-a487-a974ebf7edd1`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=371f29c2-5d87-4a5a-a487-a974ebf7edd1)
- Show all stored logs command: `$ hydra investigate show-logs 371f29c2-5d87-4a5a-a487-a974ebf7edd1`

## Logs:
- **db-cluster-371f29c2.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/db-cluster-371f29c2.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/db-cluster-371f29c2.tar.gz)
- **monitor-set-371f29c2.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/monitor-set-371f29c2.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/monitor-set-371f29c2.tar.gz)
- **loader-set-371f29c2.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/loader-set-371f29c2.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/loader-set-371f29c2.tar.gz)
- **sct-runner-371f29c2.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/sct-runner-371f29c2.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/371f29c2-5d87-4a5a-a487-a974ebf7edd1/20220907_184609/sct-runner-371f29c2.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2022.2/job/longevity/job/longevity-mv-si-4days-test/5/)

**Comment by michoecho** [comment]
2e1ec36efd29b8a78ded48ef6364e6890c8fea69 added shutdown logging to the commitlog. It was probably overlooked that hinted handoff shuts down a commitlog every 10s. At: https://github.com/scylladb/scylladb/blob/7effd4c53a5c3a7cc5e5bee79173cf4af2c84281/db/hints/manager.cc#L489

I have no idea about hints; I just followed the code up from the log message. So I don't have any fix ideas.

cc @xemul

**Comment by xemul** [comment]
@michoecho , I probably mis-understood the commitlog call sequence and put the logging into wrong place. The intention was to print something into logs when database shuts down its commitlog on scylla stop/abort. Feel free to remove this log line if you think it's not applicable there.

**Comment by kostja** [comment]
I see it in my tests as well.

**Comment by michoecho** [comment]
@xemul I think you put it in the place you intended. The problem is, the main database commitlog isn't the only commitlog in the program. Hinted handoff reuses the implemention for it's own little transient commitlogs, which are destroyed and recreated (or just shut down and reopened, I don't know how exactly it works) every 10s, and this causes the log spam.

**Comment by michoecho** [comment]
A fix (#11536) is already queued.

**Comment by avikivity** [comment]
Backported to 5.1. Earlier releases are unaffected.
